### PR TITLE
feat: add disable keyword to select and select multiple widgets

### PR DESCRIPTION
**Pull request by lp9052**
add disabled keyword to select widgets
see #144

### CODE REVIEW

**Comment by lp9052** [comment]
first PR. Want to use this to to familiarize with the process. 

Let me know if there are specific rules for this project. 

I followed instruction from development page.

Once this merged, I'll be able to contribute more.

Happy Friday.

File: solara/components/select.py [hunk 1]
```diff
@@ -96,6 +101,7 @@ def Page():
      * `values`: List of values to select from.
      * `on_value`: Callback to call when the value changes.
      * `dense`: Whether to use a denser style.
+     * `disabled`: Whether widget should allow user interact
```
Comment (maartenbreddels) [inline solara/components/select.py:104]:
```suggestion
     * `disabled`: Whether the select allow user interaction
```

File: tests/unit/select_test.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+def test_select():
+    """
+    test select widget
+    """
+    disabled = None
+    on_value = MagicMock()
+    on_value_multiple = MagicMock()
+
+    @solara.component
+    def under_test():
```
Comment (maartenbreddels) [inline tests/unit/select_test.py:17]:
```suggestion
    def Test():
```

File: tests/unit/select_test.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        nonlocal disabled
+        disabled = solara.use_reactive(False)
+        solara.Select(label="single", values=["test0", "test1", "test2"], on_value=on_value, disabled=disabled.value)
+        solara.SelectMultiple(label="multiple", values=[], all_values=["test0", "test1", "test2"], on_value=on_value_multiple, disabled=disabled.value)
+
+    _, rc = solara.render(under_test(), handle_error=False)
```
Comment (maartenbreddels) [inline tests/unit/select_test.py:23]:
```suggestion
    _, rc = solara.render(Test(), handle_error=False)
```

**Review by maartenbreddels — CHANGES_REQUESTED** [review]
Nice, thank you!

Just a few suggestions. Also in components_test.py we also have a test_select. I think we can remove that one, this one is better.

**Comment by lp9052** [comment]
Thanks Maarten!

I'll remove the test in component test.

And is there any specific reason we prefer Test() over under_test() or test()? As snakecase is the preferred convention. Pep8 will report otherwise.

File: solara/components/select.py (hunk 1 above)
Comment (lp9052) [reply solara/components/select.py:104]: will change

**Comment by lp9052** [comment]
urrr. I don't see components_test.py. Can you point out to me where is it?

**Comment by maartenbreddels** [comment]
I see it is never added to git 🙃 
But we use the CamelCase because components are special functions, they are lazy, get re-executed, and contain state, which makes them behave more like classes.

**Comment by lp9052** [comment]
just merged changes and please take a look

**Comment by maartenbreddels** [comment]
Thank you!

**Pull request merged**